server/proxy/uninstall.go

```diff
@@ -4,20 +4,28 @@
 package proxy
 
 import (
-	"fmt"
-
+	"github.com/nicksnyder/go-i18n/v2/i18n"
 	"github.com/pkg/errors"
 
 	"github.com/mattermost/mattermost-plugin-apps/apps"
 	"github.com/mattermost/mattermost-plugin-apps/server/mmclient"
 )
 
 func (p *Proxy) UninstallApp(client mmclient.Client, sessionID string, cc *apps.Context, appID apps.AppID) (string, error) {
+	loc := p.conf.I18N().GetUserLocalizer(cc.ActingUserID)
 	conf, _, log := p.conf.Basic()
 	log = log.With("app_id", appID)
 	app, err := p.store.App.Get(appID)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to get app. appID: %s", appID)
+		return "", errors.Wrap(err, p.conf.I18N().LocalizeWithConfig(loc, &i18n.LocalizeConfig{
+			DefaultMessage: &i18n.Message{
+				ID:    "apps.uninstall.error.getApp",
+				Other: "failed to get app. appID: {{.AppID}}",
+			},
+			TemplateData: map[string]string{
+				"AppID": string(appID),
+			},
+		}))
```

Comment:
Using `errors.Wrap` to return a /localized/ error might cause issues as the whole stack needs to know the users local in order to localize it's part. It's also worth noting that `GetUserLocalizer` costs one RPC call.

I was wondering if it would make sense to return a `i18n.LocalizeConfig` instead of an `error` and localize it on the top of the stack. Maybe even embedded into some kind of `UserError` struct.

Maybe my concerns aren't blocking for this PR, but I would like to double check that we are starting with the right pattern.

Reply:
We probably should have a deeper thought about this.
If were to return a special error that implements the error interface, we would be in the same situation, since this error interface would need the localizer, so the `.Error()` function can return the localized string.

If we were to return all around the LocalizeConfig, concatenating errors will be also challenging. We could return a list of localize config, and assume that it is a list of errors that have to be connected.

Another option is to fill out the localizer at the entry of the stack, and pass it along to the functions.

Thoughts?

Reply:
To be more clear: The `UserError` is had in mind won't implement `Error` for the reason you mentioned.

Concatenating errors is a good point. But maybe we shouldn't do that at all as it's more for us developers then for end users. Passing the localizer seems like a good option if we want to continue with your approach.

I like that returning something like `UserError` makes it clear that this output is for a user while that is not clear for an generic `Error`.